src/Compilers/Core/Portable/ReferenceManager/CommonReferenceManager.State.cs

```diff
@@ -485,6 +485,73 @@ internal bool IsBound
             aliasesOfReferencedAssemblies = aliasesOfReferencedAssembliesBuilder.ToImmutableAndFree();
         }
 
+        /// <summary>
+        /// Calculates map from the identities of specified symbols to the corresponding identities in the original EnC baseline metadata.
+        /// The map only includes an entry for identities that differ, i.e. for symbols representing assembly references of the current compilation that have different identities 
+        /// than the corresponding identity in baseline metadata AssemblyRef table. The key comparer of the map ignores build and revision parts of the version number, 
+        /// since these might change if the original version included wildcard.
+        /// </summary>
+        /// <param name="symbols">Assembly symbols for references of the current compilation.</param>
+        /// <param name="originalIdentities">Identities in the baseline. <paramref name="originalIdentities"/>[i] corresponds to <paramref name="symbols"/>[i].</param>
+        internal ImmutableDictionary<AssemblyIdentity, AssemblyIdentity> GetAssemblyReferenceIdentityBaselineMap(ImmutableArray<TAssemblySymbol> symbols, ImmutableArray<AssemblyIdentity> originalIdentities)
+        {
+            Debug.Assert(originalIdentities.Length == symbols.Length);
+
+            ImmutableDictionary<AssemblyIdentity, AssemblyIdentity>.Builder lazyBuilder = null;
+            for (int i = 0; i < originalIdentities.Length; i++)
+            {
+                var symbolIdentity = symbols[i].Identity;
+                var originalIdentity = originalIdentities[i];
+
+                if (originalIdentity.Version != symbolIdentity.Version)
+                {
+                    lazyBuilder = lazyBuilder ?? ImmutableDictionary.CreateBuilder<AssemblyIdentity, AssemblyIdentity>(AssemblyIdentityEqualityIgnoringBuildAndRevisionComparer.Instance);
+
+                    AssemblyIdentity anotherOriginalIdentity;
+                    if (lazyBuilder.TryGetValue(symbolIdentity, out anotherOriginalIdentity))
```

Comment:
Assuming we don't expect collisions to happen often, consider if it will be more efficient to simply attempt to ```lazyBuilder.Add(symbolIdentity, originalIdentity)``` and catch duplicate key exception.